app/models/shift_survey.rb

```diff
@@ -0,0 +1,21 @@
+class ShiftSurvey < ApplicationRecord
+  extend DateRangeFilterHelper
+
+  belongs_to :shift
+  after_create :generate_token
+  serialize :ratings, Array
+
+  def self.survey_responses_by_shift(current_user, start_at, end_at)
+    ShiftSurvey.joins(:shift)
+    .where('shifts.start_at between ? AND ?', parse_start_date(start_at), parse_end_date(end_at))
+  end
+    
+  def to_param
+    token
+  end
```

Comment:
I believe Active Admin has an issue with this? I believe this SO may have a solution: https://stackoverflow.com/questions/51671203/using-slugs-in-activeadmin-urls

(I haven't tried it, though!)